ext/php7/handlers_curl.c

```diff
@@ -92,9 +103,11 @@ static void dd_init_headers_arg(zval *arg, zval *ch) {
 
 static void dd_free_headers_arg(zval *arg) { zend_array_destroy(Z_ARRVAL_P(arg)); }
 
-static void dd_inject_distributed_tracing_headers(zval *ch) {
-    zend_string *inject_func = zend_string_init(ZEND_STRL("ddtrace\\bridge\\curl_inject_distributed_headers"), 0);
-    if (zend_hash_exists(EG(function_table), inject_func)) {
+static int dd_inject_distributed_tracing_headers(zval *ch) {
+    if (dd_inject_func == NULL) {
+        dd_inject_func = zend_string_init(ZEND_STRL("ddtrace\\bridge\\curl_inject_distributed_headers"), 0);
```

Comment:
Suggestion: use [zend_new_interned_string](https://heap.space/xref/PHP-7.0/Zend/zend_string.c?r=478f119a#24) before the requests start. Then the `dd_inject_func` string is shared between all requests and will be freed on process shutdown by the engine. Small performance boost and less lifetime code to get wrong; sounds like a win to me.